lib/sql.js

```diff
@@ -889,10 +889,26 @@ SQLConnector.prototype._constructUpdateQuery = function(model, where, fields) {
  * @param {Function} cb The callback function
  */
 SQLConnector.prototype.update = function(model, where, data, options, cb) {
+  var self = this;
   var stmt = this.buildUpdate(model, where, data, options);
-  this._executeAlteringQuery(model, stmt.sql, stmt.params, options, cb || NOOP);
+  this._executeAlteringQuery(model, stmt.sql, stmt.params, options, function(err, info) {
+    if (err) return cb(err);
+    if (where.id && info.count === 0) {
+      return cb(errorIdNotFoundForUpdate(where.id));
+    } else {
+      return cb(null, info) || NOOP;
```

Comment:
@raymondfeng @loay Why do we only return 404 when forceId is true and there are no rows that match the given ID.  Why wouldn't we also return 404 in the following cases.

forceId: false
where.name='SomeName'
No rows exist containing that name

forceId: true
where.name='SomeName'
where.id=1
No rows exist matching where clause.